src/components/oppgaver/DokumentasjonEtterspurtElementView.tsx

```diff
@@ -1,83 +1,123 @@
 import {DokumentasjonEtterspurtElement, Fil} from "../../redux/innsynsdata/innsynsdataReducer";
 import React, {useEffect, useState} from "react";
-import {
-    alertUser,
-    FileError,
-    findFilesWithError,
-    isFileErrorsNotEmpty,
-    writeErrorMessage,
-} from "../../utils/vedleggUtils";
+import {alertUser, illegalCombinedFilesSize} from "../../utils/vedleggUtils";
 import {useSelector} from "react-redux";
 import {InnsynAppState} from "../../redux/reduxTypes";
 import FileItemView from "./FileItemView";
 import AddFileButton, {TextAndButtonWrapper} from "./AddFileButton";
 import {v4 as uuidv4} from "uuid";
-import {logInfoMessage} from "../../redux/innsynsdata/loggActions";
 import {BodyShort, Label} from "@navikt/ds-react";
+import styled from "styled-components";
+import {FileValidationErrors} from "./DokumentasjonkravElementView";
+import {isFileUploadAllowed} from "../driftsmelding/DriftsmeldingUtilities";
+import useKommune from "../../hooks/useKommune";
+import {validateFile} from "./validateFile";
+import {ErrorMessageTitle} from "./ErrorMessageTitleNew";
+import ErrorMessage from "./ErrorMessage";
+import {ErrorMessage as ErrorMessageLabel} from "../errors/ErrorMessage";
+import {FormattedMessage} from "react-intl";
+
+const StyledFrame = styled.div<{hasError?: boolean}>`
+    padding: 1rem;
+    margin-top: 16px;
+    background-color: ${(props) => (props.hasError ? "var(--a-red-50)" : "var(--a-gray-200)")};
+    border-radius: 2px;
+    border-color: ${(props) => (props.hasError ? "var(--a-red-500)" : "var(--a-gray-200)")};
+    border-width: 1px;
+    border-style: solid;
+`;
 
 const DokumentasjonEtterspurtElementView: React.FC<{
     tittel: string;
     beskrivelse: string | undefined;
     oppgaveElement: DokumentasjonEtterspurtElement;
-    oppgaveElementIndex: number;
-    oppgaveId: string;
-    setOverMaksStorrelse: (overMaksStorrelse: boolean) => void;
-    onDelete: (oppgaveId: string, vedleggIndex: number, fil: Fil) => void;
-    onAddFileChange: (files: FileList, internalIndex: number, oppgaveElement: DokumentasjonEtterspurtElement) => void;
+    hendelseReferanse: string;
+    onDelete: (event: any, hendelseReferanse: string, fil: Fil) => void;
+    onAddFileChange: (event: any, hendelseReferanse: string, validFiles: Fil[]) => void;
+    setFilesHasErrors: (filesHasErrors: boolean) => void;
+    setOverMaksStorrelse: (setOverMaksStorrelse: boolean) => void;
+    overMaksStorrelse: boolean;
+    fileUploadingBackendFailed: boolean;
+    setFileUploadingBackendFailed: (setFileUploadingBackendFailed: boolean) => void;
+    filer: Fil[];
```

Comment:
Kunne vi fått til å redusere antall props her, for å gjøre komponenten litt enklere? 🤔

Reply:
Det skal være mulig å få til. 

Tenkte at propsene kan bli redusert til 3 props siden tittel, beskrivelse, oppgaveElement, hendelseReferanse og filer kan tilhører sammen, onDelete og onAddFileChange tilhører sammen, og setFilesHasErrors, setOverMaksStorrelse, overMaksStorrelse, fileUploadingBackendFailed og setFileUploadingBackendFailed tilhører sammen.

Men jeg tenker at dette kan bli gjort med neste refaktorering.